lib/SPIRV/SPIRVToLLVMDbgTran.h

```diff
@@ -77,10 +77,11 @@ class SPIRVToLLVMDbgTran {
                 SPIRVEIS_NonSemantic_Shader_DebugInfo_200) &&
            "Unexpected extended instruction set");
     auto It = DebugInstCache.find(DebugInst);
-    if (It != DebugInstCache.end())
+    if (It != DebugInstCache.end() && It->second != nullptr)
       return static_cast<T *>(It->second);
     MDNode *Res = transDebugInstImpl(DebugInst);
-    DebugInstCache[DebugInst] = Res;
+    if (Res)
+      DebugInstCache[DebugInst] = Res;
     return static_cast<T *>(Res);
```

Comment:
Why so that we check `It->second` for nullptr before casting, but we are ok with `Res` can be nullptr as a return value? Or is it the subtle logic of debug instructions nested translation?

Reply:
It's a leftover from a previous version of the implementation. Removed.